### DISCUSSION TITLE
Ban with IP message

### DISCUSSION THREAD

**Discussion by PH5HP** [rejetto/hfs#368] [Ideas]
Hi,

I would like a ban reason if I block an IP address. Also a time limiet would be nice. Maybe per ban adjustable? I have some rules about downloading and some people still try to download 157Gb in one ZIP file and that is against my rules for downloading. :-)

Kind regards,
Henk - ftp.ph5hp.nl

**Comment by rejetto** [comment]
do you need to block them because they don't follow the rules? 
then let's try to prevent it :)
let's see if this new dedicated permission works
<img width="337" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/b72073d3-cfbb-4b9a-ba93-c8a33c2fd52b">

i made some tests, but please test yourself and LMK.
This is if you use the exe version [hfs-windows.zip](https://github.com/rejetto/hfs/files/12790325/hfs-windows.zip)

**Reply by rejetto** [reply]
just replaced the zip with a better one

**Comment by PH5HP** [comment]
Hi,

It works, but is not an elegant way tp prevent zipping large amounts of data. If I am correct a file is still created (22 bytes) and it's empty. More elegant would be to hide the ZIP button or a message that ZIP is not allowed.

Kind regards,
Henk - ftp.ph5hp.nl

**Comment by rejetto** [comment]
sure, i forgot to make changes to that part.
hopefully alpha2 will solve it.
https://github.com/rejetto/hfs/releases/tag/v0.49.0-alpha2

**Reply by PH5HP** [reply]
Hi,

Thank you. Much better this way. Another idea would be if someone does not want to disable the ZIP function that we could define a maximum amount of DATA which is allowed to ZIP. Just a thought from myside.

Kind regards,
Henk - ftp.ph5hp.nl

**Reply by rejetto** [reply]
it would be nice, but hfs doesn't know the amount of data.
Knowing that size could take many seconds, even minutes, before starting.

**Reply by PH5HP** [reply]
Hi,

You are right about that. I overlooked that. On the other hand most of the files are already compressed om my server and I could define now the directories in which ZIPPING the data is allowed. 

Kind regards,
Henk - ftp.ph5hp.nl

**Reply by Saboreq** [reply]
@rejetto how can i update my hfs? i was trying update 0.49.0-alpha2 but this doesnt work :/ 
![image](https://github.com/rejetto/hfs/assets/118023809/bc4270ba-7d9b-4463-a30b-42bf5134da4a)

**Comment by rejetto** [comment]
@Saboreq 
i don't fully understand your screenshot. You cut a bit too much.
What is this?
<img width="244" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/d38a03a7-56a7-4472-83b9-aa8d0e72a74c">
something you typed in the console?

**Reply by Saboreq** [reply]
![image](https://github.com/rejetto/hfs/assets/118023809/9b1765b1-a638-4bff-9033-56705709a7d8)

**Comment by rejetto** [comment]
i see. Actually, "update" command at the moment doesn't let you pick the version, so the version you typed is ignored.
I will change this, soon.
i just tried to update the same way you did and got no error.
So, what happens if you try again?

**Reply by Saboreq** [reply]
![image](https://github.com/rejetto/hfs/assets/118023809/47d2b78f-d9ea-4750-af59-72b66b976ea3)
im getting this, btw will i lose my all "config" if i will just replace .exe file with latest one?

**Comment by rejetto** [comment]
Replacing the exe is safe

**Comment by rejetto** [comment]
@Saboreq i couldn't think of a reason why you get the error.
did you succeed updating otherwise?

**Comment by rejetto** [comment]
@Saboreq try updating using the browser, Admin-panel > check for updates

**Reply by Saboreq** [reply]
sorry for long respond, i just replaced .exe and it worked prob using admin panel should work too

**Reply by Saboreq** [reply]
Also i managed to make my site https://saboreq.com work with https://files.saboreq.com i had to make DNS changes which i didint understand but now its working B))

**Reply by rejetto** [reply]
congratulations, it works well

**Reply by Saboreq** [reply]
did u checked them? what do u think? :D

**Reply by rejetto** [reply]
think about what? your server is working well. Did you generate the certificate using hfs ?

**Reply by Saboreq** [reply]
yes i generated certificate using hfs i meant what u think about "style or design" my sites saboreq.com and saboreq.xyz (.com design from my friend and .xyz public source)

**Reply by rejetto** [reply]
cool design, but i would get rid of all the "click to...." intermediate screens
